src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,39 +1,83 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { clearCurrentTodo } from '../../features/currentTodo';
+import { useAppDispatch, useAppSelector } from '../../hooks';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setLoading] = useState(false);
+
+  useEffect(() => {
+    if (!currentTodo?.userId) {
+      return;
+    }
+
+    async function getUserData() {
+      try {
+        setLoading(true);
+        const userData = await getUser(currentTodo.userId);
+
+        setUser(userData); // Store in component state, not Redux
+      } catch (e) {
+        setUser(null);
+      } finally {
+        setLoading(false);
+      }
+    }
+
+    getUserData();
+  }, [currentTodo]);
+
+  if (!currentTodo) {
+    return null;
+  }
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {isLoading && <Loader />}
 
       <div className="modal-card">
         <header className="modal-card-head">
           <div
             className="modal-card-title has-text-weight-medium"
             data-cy="modal-header"
           >
-            Todo #3
+            {`Todo #${currentTodo?.id}`}
```

Comment:
```suggestion
            {`Todo #${currentTodo.id}`}
```
You've already checked `currentTodo` above. Same everywhere below